pkg/controller/clusterclaim/clusterclaim_controller.go

```diff
@@ -117,20 +114,37 @@ func claimForClusterDeployment(cd *hivev1.ClusterDeployment) *types.NamespacedNa
 	}
 }
 
-func requestsForClusterDeployment(ctx context.Context, o client.Object) []reconcile.Request {
-	cd, ok := o.(*hivev1.ClusterDeployment)
-	if !ok {
-		return nil
-	}
+func requestsForClusterDeployment(ctx context.Context, cd *hivev1.ClusterDeployment) []reconcile.Request {
+	// FIXME how to handle when object is not a ClusterDeployment?
```

Comment:
IIUC that is no longer possible. Certainly at this level, as the param is now typed. I haven't dug into the upstream changes, but I'm guessing they're doing the type checking in the library now, so we don't have to do it here.

Reply:
Do these comments need to go away now?

Reply:
Yes, the type system has our backs here